### ISSUE TITLE
Illegal client ID

### ISSUE THREAD

**Issue by iridescent-aeon** [sentinel-hub/eo-learn#240]
I'm trying to run the RequestDataComparison notebook. I've set up the configuration in the following way:
1) config.instance_id from here (field in red):
![sentinel1](https://user-images.githubusercontent.com/64409828/89031855-9b44ed80-d333-11ea-890c-7db7cea47ea1.png)

2) config.sh_client_id (field in green) and config.sh_client_secret (field in purple) from here:
![sentinel2](https://user-images.githubusercontent.com/64409828/89032184-68e7c000-d334-11ea-9f29-c24d67f04925.png)

I am getting the following error:
![code](https://user-images.githubusercontent.com/64409828/89032840-c9c3c800-d335-11ea-8749-dc00be34c199.png)

Where am I going wrong?

**Comment by AleksMat** [comment]
Hi @iridescent-aeon,

It turns out you took client id and secret from the wrong places.

- The ID in the green field is a user ID and is not important for making API requests.
- The ID in the purple field is an OAuth client ID and should be assigned to `config.sh_client_id`.
- An OAuth client secret is a string of characters that you get when you make a new OAuth client. You can only see it the first time when you create a new client and never again. But if you ever lose it you can always simply create a new client and delete an old one. That string should be assigned to `config.sh_client_secret`.

Everything else you described seems to be ok.

**Comment by iridescent-aeon** [comment]
Interesting, I can swear I tried the option you suggested, but maybe there are problems with vscodium + jupyter notebooks, or something else.

tldr; changed appropriately and reopened vscodium, now everything works.
Thank you for the input :)

**Comment by likhitha-d01** [comment]
> Hi @iridescent-aeon,
> 
> It turns out you took client id and secret from the wrong places.
> 
> * The ID in the green field is a user ID and is not important for making API requests.
> * The ID in the purple field is an OAuth client ID and should be assigned to `config.sh_client_id`.
> * An OAuth client secret is a string of characters that you get when you make a new OAuth client. You can only see it the first time when you create a new client and never again. But if you ever lose it you can always simply create a new client and delete an old one. That string should be assigned to `config.sh_client_secret`.
> 
> Everything else you described seems to be ok.

I changed the config exactly according to the instructions. Where am I going wrong?

**Comment by mlubej** [comment]
> > Hi @iridescent-aeon,
> > It turns out you took client id and secret from the wrong places.
> > 
> > * The ID in the green field is a user ID and is not important for making API requests.
> > * The ID in the purple field is an OAuth client ID and should be assigned to `config.sh_client_id`.
> > * An OAuth client secret is a string of characters that you get when you make a new OAuth client. You can only see it the first time when you create a new client and never again. But if you ever lose it you can always simply create a new client and delete an old one. That string should be assigned to `config.sh_client_secret`.
> > 
> > Everything else you described seems to be ok.
> 
> I changed the config exactly according to the instructions. Where am I going wrong?

Dear @likhitha-d01, please provide more information. Did you follow the steps [here](https://sentinelhub-py.readthedocs.io/en/latest/configure.html)? Check for typos/empty spaces in the configuration file. Be sure to use double quotes in the json.

**Comment by likhitha-d01** [comment]
Hi @mlubej. Thanks for the fast reply. 

This is the output of `sentinelhub.config --show`
![abc](https://user-images.githubusercontent.com/34205126/94059803-9c107200-fe00-11ea-9d3f-2060c7b1ec04.png)

The instance_id ( the one in blue ), is the id of the configuration in the Configuration utility.
![Uploading Screenshot from 2020-09-24 00-45-51.png…]()

The sh_client_id ( the one in green ) is the id of the OAuth Client in User Settings
![Screenshot from 2020-08-18 10-03-53](https://user-images.githubusercontent.com/34205126/94060285-543e1a80-fe01-11ea-8660-9ee2b08559a4.png)

The secret_id is the one we get during creating the OAuth client.

A bit of background: 
I am running the Land Use Land Cover example notebook, in a VM in GCP. I'm hosting the notebook using the jupyter-notebook.
Also, I have initialized the parameters in the command line.
I also checked for spaces and typos. 

Thanks again for your help!

**Comment by likhitha-d01** [comment]
> > > Hi @iridescent-aeon,
> > > It turns out you took client id and secret from the wrong places.
> > > 
> > > * The ID in the green field is a user ID and is not important for making API requests.
> > > * The ID in the purple field is an OAuth client ID and should be assigned to `config.sh_client_id`.
> > > * An OAuth client secret is a string of characters that you get when you make a new OAuth client. You can only see it the first time when you create a new client and never again. But if you ever lose it you can always simply create a new client and delete an old one. That string should be assigned to `config.sh_client_secret`.
> > > 
> > > Everything else you described seems to be ok.
> > 
> > 
> > I changed the config exactly according to the instructions. Where am I going wrong?
> 
> Dear @likhitha-d01, please provide more information. Did you follow the steps [here](https://sentinelhub-py.readthedocs.io/en/latest/configure.html)? Check for typos/empty spaces in the configuration file. Be sure to use double quotes in the json.

Hi @mlubej. Thanks for the fast reply. 

This is the output of `sentinelhub.config --show`
![abc](https://user-images.githubusercontent.com/34205126/94059803-9c107200-fe00-11ea-9d3f-2060c7b1ec04.png)

The instance_id ( the one in blue ), is the id of the configuration in the Configuration utility.
![Screenshot from 2020-09-24 00-45-51](https://user-images.githubusercontent.com/34205126/94061189-acc1e780-fe02-11ea-8659-5cc99bad038c.png)

The sh_client_id ( the one in green ) is the id of the OAuth Client in User Settings
![Screenshot from 2020-08-18 10-03-53](https://user-images.githubusercontent.com/34205126/94060285-543e1a80-fe01-11ea-8660-9ee2b08559a4.png)

The secret_id is the one we get during creating the OAuth client.

A bit of background: 
I am running the Land Use Land Cover example notebook, in a VM in GCP. I'm hosting the notebook using the jupyter-notebook.
Also, I have initialized the parameters in the command line.
I also checked for spaces and typos. 

Thanks again for your help!

**Comment by mlubej** [comment]
Hi @likhitha-d01,

everything seems to make sense to me. My next idea is to check the output of `!sentinelhub.config --show` in the notebook where you are also running things. It is possible that the JN kernel is using the wrong instance of the python interpreter.

Besides checking `!sentinelhub.config --show` in the JN cell, you can also try

```python
from sentinelhub import SHConfig
SHConfig()
```
and check the output. Is it what you expect it to be?

**Comment by likhitha-d01** [comment]
Hi @mlubej 

The output is the same as `!sentinelhub.config --show`. 
Could it be an issue with any other parameters? or maybe the following code:

```python
band_names = ['B02', 'B03', 'B04', 'B08', 'B11', 'B12']
add_data = SentinelHubInputTask(
    bands_feature=(FeatureType.DATA, 'BANDS'),
    bands = band_names,
    resolution=10,
    maxcc=0.8,
    time_difference=datetime.timedelta(minutes=120),
    data_source=DataSource.SENTINEL2_L1C,
    additional_data=[(FeatureType.MASK, 'dataMask', 'IS_DATA'),
                     (FeatureType.MASK, 'CLM'),
                     (FeatureType.DATA, 'CLP')],
    max_threads=5,
    #instance_id="e7d4aa46-4893-4b2b-8d54-d70998e4e006"
)

# TASK FOR SAVING TO OUTPUT (if needed)
path_out = './eopatches/'
if not os.path.isdir(path_out):
    os.makedirs(path_out)
save = SaveTask(path_out, overwrite_permission=OverwritePermission.OVERWRITE_PATCH)

workflow = LinearWorkflow(
    add_data,
    save
)

# Execute the workflow
time_interval = ['2019-01-01', '2019-12-31'] # time interval for the SH request

# define additional parameters of the workflow
execution_args = []
for idx, bbox in enumerate(bbox_list[patchIDs]):
    execution_args.append({
        add_data:{'bbox': bbox, 'time_interval': time_interval},
        save: {'eopatch_folder': f'eopatch_{idx}'}
    })
    
executor = EOExecutor(workflow, execution_args, save_logs=True)
executor.run(workers=5, multiprocess=True)

executor.make_report()

```

**Comment by iridescent-aeon** [comment]
Hi @likhitha-d01,

I don't think your problem comes from the code or using wrong ids, but rather your configuration in the configuration utility. Most likely some of the bands you are using in the code were not included in the "custom_script" configuration on the sentinel hub page. I have stopped using this for a while, so perhaps someone else would be able to help you adjust the configuration, based on your required bands. Alternatively, if my memory serves me right, there was a config that included all bands, from the pre-given templates.

**Comment by mlubej** [comment]
> Most likely some of the bands you are using in the code were not included in the "custom_script" configuration on the sentinel hub page.

This likely isn't it, since we implemented changes with the processing API to bypass this completely.

@likhitha-d01 it seems to me that what you're doing is correct. Perhaps there was some mismatch along the way. I would suggest deleting your `oauth` client and creating a new one. Let me know if this works.

**Comment by likhitha-d01** [comment]
I tried the same configuration with the Sentinel FIS, and it worked perfectly fine. I guess there is some issue with the `SentinelHubInputTask` function

Also, @iridescent-aeon thanks for the Google Engine tip!

**Comment by mlubej** [comment]
Hi @likhitha-d01,

FIS request uses instance ID only and not oauth-client, so this is not the reason either, unfortunately.

May I have your email account, so that I can check your setup in our system?

**Comment by likhitha-d01** [comment]
Hello,

likhithabhu1999@gmail.com is my email account registered on the sentinel
hub website.

Thanks,
Likhitha

**Comment by likhitha-d01** [comment]
Hi @mlubej 

It worked when I passed the `config = SHConfig()` to the `SentinelHubInputTask` function explicitly.

I ran the same example **Land Use Land Cover** notebook.

Thanks for your help :)

**Comment by mlubej** [comment]
Hi @likhitha-d01 

sorry for not being able to help sooner, but I'm glad to hear this was resolved. 

Thinking about it now, it might also be possible that you have some other instance of sh-py installed and the config file was changed in the one not used in the notebook. You could try removing all installations of sh-py in case you have more than one and then reinstalling.

Passing the config file as you did now proves that the credentials are correct.

Anyway, glad it works now :)
